problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -7,30 +7,118 @@ function MentoringWithStepsBlock(runtime, element) {
         var blockType = $(child.element).data('block-type');
         if (blockType === 'sb-step') {
-            steps.push(child);
+            registerStep(child);
         }
     }
 
-    var activeStep = $('.mentoring', element).data('active-step');
+    var activeStepIndex = $('.mentoring', element).data('active-step');
     var attemptsTemplate = _.template($('#xblock-attempts-template').html());
     var message = $('.sb-step-message', element);
     var checkmark, submitDOM, nextDOM, reviewButtonDOM, tryAgainDOM,
         gradeDOM, attemptsDOM, reviewLinkDOM, submitXHR;
     var reviewStepDOM = $("div.xblock[data-block-type=sb-review-step], div.xblock-v1[data-block-type=sb-review-step]", element);
     var hasAReviewStep = reviewStepDOM.length == 1;
+    
+    /**
+     * Registers step to in this builder, it stores the xblock element, and some additional metadata used to safely
+     * remove and re-attach xblock to the DOM. See showStep and hideStep to see why we need to do it.
+     * @param step xblock instance to register
+     */
+    function registerStep(step) {
+        var $element = $(step.element);
+        var $anchor = $('<span class="xblock-sb-anchor"/>');
+        $anchor.insertBefore($element);
+        var step_wrapper = {
+            idx: step.length,
+            $element: $element,
+            xblock: step,
+            $anchor: $anchor
+        };
+        steps.push(step_wrapper);
+    }
+
+    /**
+     * Returns wrapper for active step
+     * @returns {*}, an object containing xblock, and additional metadata (see registerStep where this object is created
+     * for properties)
+     */
+    function getWrapperForActiveStep() {
+        return steps[activeStepIndex];
+    }
+
+    /**
+     * Returns active step)
+     * @returns {*}
+     */
+    function getActiveStep() {
+        return getWrapperForActiveStep().xblock;
+    }
+
+    /**
+     * Calls a function for each registered step. Object passed to this function is an step wrapper object
+     * (see registerStep where this object is created a list of properties)
+     *
+     * @param func single arg function.
+     */
+    function forEachStep(func){
+        for (var idx=0; idx<steps.length; idx++) {
+            func(steps[idx]);
+        }
+    }
+
+    /**
+     * Displays a step
+     * @param step_wrapper 
+     */
+    function showStep(step_wrapper) {
+        step_wrapper.$element.show();
+        step_wrapper.xblock.updateChildren();
+    }
+
+    /**
+     * Hides a step
+     * @param step_wrapper
+     */
+    function hideStep(step_wrapper) {
+        // This is a hacky workaround, but it works. It detaches this element from dom, and re-attaches it immediately,
+        // which has the side effect of stopping any video elements (tested with Ooyala and Youtube). 
+        // This solution was chosen as permanently detaching xblocks from DOM is not anticipated by lot of software 
+        // and some code in this repository. 
```

Comment:
@jbzdak "from dom, and" ⟶ "from the DOM and"

What do you mean by "lot of software"? Might be worth specifying which software :)